llvm/test/Transforms/LoopVectorize/single_early_exit_live_outs.ll

```diff
@@ -1480,57 +1480,21 @@ define i64 @same_exit_block_pre_inc_use1_reverse() {
 ; CHECK-NEXT:    [[P2:%.*]] = alloca [1024 x i8], align 1
 ; CHECK-NEXT:    call void @init_mem(ptr [[P1]], i64 1024)
 ; CHECK-NEXT:    call void @init_mem(ptr [[P2]], i64 1024)
-; CHECK-NEXT:    br i1 false, label [[SCALAR_PH:%.*]], label [[VECTOR_PH:%.*]]
```

Comment:
This looks like a regression to me. Are you suggesting that we simply cannot ever check for dereferenceability in reverse loops for some fundamental reason or that the test itself was invalid? If it's the latter I'm happy to rewrite the test. :) It would be a shame to lose this functionality.

Reply:
The functionality for supporting testing whether loads could be dereferenced in reverse loops was added in https://github.com/llvm/llvm-project/pull/96752 specifically to support such cases.

Reply:
Yep the original version wasn't handling this, but should be fixed in the latest version.